### PR TITLE
feat: catch panics in identify

### PR DESCRIPTION
**Pull request by Stebalien**
We haven't seen any panics in identify, but I'm going through and adding panic handlers to libp2p handlers because the benefits often outweigh the risk:

- Risk: We might continue with corrupted state. In this case, that means continuing with a corrupted peerstore, which likely doesn't have a huge impact on anything. If something is really wrong, we'll crash somewhere else.
- Benefit: if we panic when processing an identify response, we won't crash the entire node.

Part of #1389

### CODE REVIEW

File: p2p/protocol/identify/id.go [hunk 1]
```diff
@@ -421,16 +421,32 @@ func (ids *idService) sendIdentifyResp(s network.Stream) {
...
+
+		// Cleanup by resetting on any kind of failure.
+		if err != nil {
+			s.Reset()
+		}
+	}()
+
 	if err := s.Scope().SetService(ServiceName); err != nil {
 		log.Warnf("error attaching stream to identify service: %s", err)
-		s.Reset()
```
Comment (marten-seemann) [inline p2p/protocol/identify/id.go:427]: You're creating a new `err` here that shadows the `err` return value, don't you?

Comment (Stebalien) [reply p2p/protocol/identify/id.go:427]: `err` will be shadowed inside this condition, but the `err` in the "defer" will still refer to the named return value.

**Comment by BigLep** [comment]
2022-09-16 conversation: we need to evaluate if this is the right place to do this or determine if our panic handlers in serialization functions are sufficient.

**Comment by marten-seemann** [comment]
Closing since we've done a major refactor of Identify.

**Pull request closed without merging**